File: ciphertoken/algorithms.py
"""Python wrapper for Rust algorithms submodule."""

from .ciphertoken import algorithms as _algorithms

HS256 = _algorithms.HS256
HS384 = _algorithms.HS384
HS512 = _algorithms.HS512
RS256 = _algorithms.RS256
RS384 = _algorithms.RS384
RS512 = _algorithms.RS512
ES256 = _algorithms.ES256
ES384 = _algorithms.ES384
PS256 = _algorithms.PS256
PS384 = _algorithms.PS384
PS512 = _algorithms.PS512
EDDSA = _algorithms.EDDSA

__all__ = [
    "HS256",
    "HS384",
    "HS512",
    "RS256",
    "RS384",
    "RS512",
    "ES256",
    "ES384",
    "PS256",
    "PS384",
    "PS512",
    "EDDSA",
]

File: src/algorithms.rs
use pyo3::prelude::*;

pub fn register_algorithms_module(py: Python) -> PyResult<Py<PyModule>> {
    let alg_module = PyModule::new(py, "algorithms")?;

    // HMAC
    alg_module.add("HS256", "HS256")?;
    alg_module.add("HS384", "HS384")?;
    alg_module.add("HS512", "HS512")?;

    // RSA
    alg_module.add("RS256", "RS256")?;
    alg_module.add("RS384", "RS384")?;
    alg_module.add("RS512", "RS512")?;

    // ECDSA
    alg_module.add("ES256", "ES256")?;
    alg_module.add("ES384", "ES384")?;

    // RSA-PSS
    alg_module.add("PS256", "PS256")?;
    alg_module.add("PS384", "PS384")?;
    alg_module.add("PS512", "PS512")?;

    // Edwards Curve
    alg_module.add("EDDSA", "EdDSA")?;

    Ok(alg_module.into())
}
